integration/mdc-migration/sample-project/tsconfig.app.json

```diff
@@ -5,11 +5,7 @@
     "outDir": "./out-tsc/app",
     "types": []
   },
-  "files": [
-    "src/main.ts",
-    "src/polyfills.ts"
-  ],
-  "include": [
-    "src/**/*.d.ts"
-  ]
+  "files": ["src/main.ts", "src/polyfills.ts"],
+  "include": ["src/**/*.ts", "src/**/*.d.ts"],
+  "exclude": ["src/**/*.spec.ts"]
```

Comment:
Is this change necessary? looking at the default config the CLI generates, this shouldn't be needed.

Reply:
Without this change, the typescript files (and by association the HTML files) were not being picked up to be migrated (https://github.com/angular/components/blob/main/src/cdk/schematics/update-tool/index.ts#L67)

Reply:
That would be unexpected though because all transitive files references by `main.ts` will be part of the `ts.Program` source files. A real CLI application does not have the setup like that, so I think the integration test should not be adjusted in that way. WDYT?

Reply:
Yeah [main.ts](https://github.com/angular/components/blob/afa312bff0a8b979da8d08280bcf2dace6e85dde/integration/mdc-migration/sample-project/src/main.ts) gets picked up in the source files. Although it stops there since it doesn't have a class decorator (which [the resource collector class uses to populate the list of resolved templates and stylesheets](https://github.com/angular/components/blob/main/src/cdk/schematics/update-tool/component-resource-collector.ts#L51)). Stylesheets are still getting migrated because there is an [additional search for stylesheets](https://github.com/angular/components/blob/main/src/cdk/schematics/update-tool/index.ts#L120) (meant for global stylesheets) that picks up the components specific ones.

Although if applications don't have tsconfig files like this, then it shouldn't be changed.

Reply:
Before it actually hits the component resource collector, the `ts.Program` analyzes the `main.ts` file and finds all transitively imported files. Then for every file it runs the resource collector. It sounds like the expected test files are somehow not included transitively?